### DISCUSSION TITLE
Retrieve Cell Values

### DISCUSSION THREAD

**Discussion by rileyhawk1417** [palexdev/MaterialFX#159] [Archived]
Hi can anyone share an example where they you retrieve cell values from a table or list (e.g Name).
Been trying to use the getSelection() property but nothing only got an object in return.

This is what I use when using plain JavaFX.

```java
    public void deleteRow() {
        ObservableList<MemberModel> allRows, selectedRow;
        allRows = psqlTable.getItems();
        String id_ = psqlTable.getSelectionModel().getSelectedItem().getID();

        sqlite.delete_row_by_id(id_);
        selectedRow.forEach(allRows::remove);
    }
```

This is the current piece of code am working on with MaterialFX (MFXListView), from here all am getting is the MapProperty.
Theres nothing much to do with that afterwards besides adding or removing.
```java
    private void deleteSelected(){
        ObservableList<UsersModel> selectedRow;
         //Even toString() does not bring out the values.
        selectedRow = userList.getSelectionModel().selectionProperty().values();

        System.out.println("ID row Selected: " + selectedRow);
    }
```

Any pointers or advice will be useful.

**Comment by palexdev** [comment]
Lists and tables use a Map since the selection can contain multiple items
Once you retrieve the selection List you have to get the item you want to remove

Or you can remove all the selected items, the code should be very similar to the JavaFX one

**Reply by rileyhawk1417** [reply]
Well managed to play around with some functions and found something that works.
Not sure if theres a shorter way to do it.

```java
ObservableMap<Integer, UsersModel> listValues = userList.getSelectionModel().getSelection();
ObservableList<UsersModel> namesList = FXCollections.observableArrayList(listValues.values());

namesList.listIterator().next().getUserName();
```

This retrieves the Map value on the clicked cell, had paired this with a delete button. Might try and use it somewhere else.

Used  this [reference](https://stackoverflow.com/questions/53349542/javafx-observablelist-as-a-value-in-a-map) from Stack Overflow, hopefully it helps someone.

**Reply by palexdev** [reply]
@rileyhawk1417 I opened an issue for this. Improvements must be made.

**Reply by rileyhawk1417** [reply]
Ok am not much of a Java Expert but can pitch in and offer some suggestions and try test things out.

**Reply by palexdev** [reply]
I used a Map since every index is unique and makes retrieval of values per index really easy
On the other hand though Maps return a generic Collection when calling values() and that could be a problem
I have to thing of something that still ensures the mapping index -> item but makes it easier to retrieve items
